Documentation/guides/OneDotNet.md

```diff
@@ -135,6 +135,36 @@ Default Android related file globbing behavior is defined in `Microsoft.Android.
 This behavior can be disabled for Android items by setting `$(EnableDefaultAndroidItems)` to `false`, or
 all default item inclusion behavior can be disabled by setting `$(EnableDefaultItems)` to `false`.
 
+## Linker (ILLink)
+
+.NET 5 and higher has new [settings for the linker][linker]:
+
+* `<PublishTrimmed>true</PublishTrimmed>`
+* `<TrimMode>copyused</TrimMode>` - Enable assembly-level trimming.
+* `<TrimMode>link</TrimMode>` - Enable member-level trimming.
+
+In Android application projects by default, `Debug` builds will not
+use the linker and `Release` builds will set `PublishTrimmed=true` and
+rely on the default `TrimMode=CopyUsed` from the dotnet/sdk.
+
+If the legacy `AndroidLinkMode` setting is used, both `SdkOnly` and
+`Full` will default to equivalent linker settings:
+
+* `<PublishTrimmed>true</PublishTrimmed>`
+* `<TrimMode>link</TrimMode>`
+
+With these settings only BCL and SDK assemblies marked with
+`%(Trimmable)` will be linked at the member level. There is not an
+exact equivalent of `AndroidLinkMode=Full`, but a custom MSBuild
```

Comment:
I came up with this: https://github.com/xamarin/xamarin-macios/blob/1463967d003ad594c140489388fb1f906a0a9565/dotnet/targets/Xamarin.Shared.Sdk.targets#L208-L214 (which can probably be improved to only set TrimMode for assemblies that hasn't set it already) , because it's quite common for our customers to want to use Full to link away as much as possible (i.e. link every single assembly). Apple has app size limits which can be quite annoying... and a common first approach is to link everything and hope both that the app is small enough for Apple and continues to work (and it works often enough)

Reply:
I was thinking we would deprecate `$(AndroidLinkMode)` eventually, but I'll change our `.targets` to do this for now.

There isn't even a UI to setup most of these things in .NET 5 yet, you only get a checkbox that does `PublishTrimmed=true`:

![image](https://user-images.githubusercontent.com/840039/93618314-b4196900-f99c-11ea-8f11-2421dae5bcf2.png)

Reply:
That UI will have to change for Xamarin...